src/vm/hints/blake2s_hash.rs

```diff
@@ -0,0 +1,335 @@
+const IV: [u64; 8] = [
+    0x6A09E667, 0xBB67AE85, 0x3C6EF372, 0xA54FF53A, 0x510E527F, 0x9B05688C, 0x1F83D9AB, 0x5BE0CD19,
+];
+
+const SIGMA: [[usize; 16]; 10] = [
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15],
+    [14, 10, 4, 8, 9, 15, 13, 6, 1, 12, 0, 2, 11, 7, 5, 3],
+    [11, 8, 12, 0, 5, 2, 15, 13, 10, 14, 3, 6, 7, 1, 9, 4],
+    [7, 9, 3, 1, 13, 12, 11, 14, 2, 6, 5, 10, 4, 0, 15, 8],
+    [9, 0, 5, 7, 2, 4, 10, 15, 14, 1, 11, 12, 6, 8, 3, 13],
+    [2, 12, 6, 10, 0, 11, 8, 3, 4, 13, 7, 5, 15, 14, 1, 9],
+    [12, 5, 1, 15, 14, 13, 4, 10, 0, 7, 6, 3, 9, 2, 8, 11],
+    [13, 11, 7, 14, 12, 1, 3, 9, 5, 0, 15, 4, 8, 6, 2, 10],
+    [6, 15, 14, 9, 11, 3, 0, 8, 12, 2, 13, 7, 1, 4, 10, 5],
+    [10, 2, 8, 4, 7, 6, 1, 5, 15, 11, 9, 14, 3, 12, 13, 0],
+];
+
+fn right_rot(value: u64, n: u64) -> u64 {
+    (value >> n) | ((value & (2_u64.pow(n as u32) - 1)) << (32 - n))
+}
+
+fn mix(a: u64, b: u64, c: u64, d: u64, m0: u64, m1: u64) -> (u64, u64, u64, u64) {
+    let a = (a + b + m0) % 2_u64.pow(32);
+    let d = right_rot(d ^ a, 16);
+    let c = (c + d) % 2_u64.pow(32);
+    let b = right_rot(b ^ c, 12);
+    let a = (a + b + m1) % 2_u64.pow(32);
+    let d = right_rot(d ^ a, 8);
+    let c = (c + d) % 2_u64.pow(32);
+    let b = right_rot(b ^ c, 7);
```

Comment:
```suggestion
    let a = (a + b + m0) & u32::MAX as u64;
    let d = right_rot(d ^ a, 16);
    let c = (c + d) & u32::MAX as u64;
    let b = right_rot(b ^ c, 12);
    let a = (a + b + m1) & u32::MAX as u64;
    let d = right_rot(d ^ a, 8);
    let c = (c + d) & u32::MAX as u64;
    let b = right_rot(b ^ c, 7);
```

Reply:
As an explanation for the suggested patch: binary and with `2^n - 1` is the same as modulus with `2^n` for unsigned arithmetic. In general, dividing by a power of your base is shifting places, and the remainder is necessarily what you would lose after that shift, that is whatever is in a position to the left of the power itself. That set to all `1` is also `2^n - 1`, which is the maximum number you can represent with `n` bits.
So, `x % 2.pow(32) == x & (2.pow(32) - 1) == x & u32::MAX`.

Reply:
`&` with a constant is much more efficient for the computer to perform than division.

Reply:
Thanks for the explanation! applied change!